include/zephyr/drivers/mm/rat.h

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright (c) 2023 Texas Instruments Incorporated
+ * Copyright (c) 2023 L Lakshmanan
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_RAT_H_
+#define ZEPHYR_INCLUDE_RAT_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <stdint.h>
+
+#define ADDR_TRANSLATE_MAX_REGIONS (16u)
+#define RAT_CTRL(base_addr, i)     (volatile uint32_t *)(base_addr + 0x20 + 0x10 * (i))
+#define RAT_BASE(base_addr, i)     (volatile uint32_t *)(base_addr + 0x24 + 0x10 * (i))
+#define RAT_TRANS_L(base_addr, i)  (volatile uint32_t *)(base_addr + 0x28 + 0x10 * (i))
+#define RAT_TRANS_H(base_addr, i)  (volatile uint32_t *)(base_addr + 0x2C + 0x10 * (i))
+#ifndef NULL
+#define NULL (0)
+#endif
+
+/**
+ * \brief Enum's to represent different possible region size for the address translate module
+ */
+typedef enum address_trans_region_size_e {
+	address_trans_region_size_1 = 0x0,
+	address_trans_region_size_2,
+	address_trans_region_size_4,
+	address_trans_region_size_8,
+	address_trans_region_size_16,
+	address_trans_region_size_32,
+	address_trans_region_size_64,
+	address_trans_region_size_128,
+	address_trans_region_size_256,
+	address_trans_region_size_512,
+	address_trans_region_size_1K,
+	address_trans_region_size_2K,
+	address_trans_region_size_4K,
+	address_trans_region_size_8K,
+	address_trans_region_size_16K,
+	address_trans_region_size_32K,
+	address_trans_region_size_64K,
+	address_trans_region_size_128K,
+	address_trans_region_size_256K,
+	address_trans_region_size_512K,
+	address_trans_region_size_1M,
+	address_trans_region_size_2M,
+	address_trans_region_size_4M,
+	address_trans_region_size_8M,
+	address_trans_region_size_16M,
+	address_trans_region_size_32M,
+	address_trans_region_size_64M,
+	address_trans_region_size_128M,
+	address_trans_region_size_256M,
+	address_trans_region_size_512M,
+	address_trans_region_size_1G,
+	address_trans_region_size_2G,
+	address_trans_region_size_4G
+} address_trans_region_size;
+
+/**
+ * \brief Region config structure
+ */
+typedef struct address_trans_region_config_ {
+
+	uint64_t system_addr;	/* translated 48b system addr as seen by the SOC, MUST align to
```

Comment:
if this is going to be a generic driver, then does it really make sense to say it will be a "48b system addr" always?
I thought this was the case only in TI AM625 M4 core's RAT.

If in future a new device wanted to use RAT does it necessarily have to be a 48bit addr or it can be anything else too?

Reply:
It can be anything as long as the mapping function accommodates for it, and the mapping function can be custom too. Will remove the comment.